fortitude/src/check.rs

```diff
@@ -323,17 +328,21 @@ fn get_files<P: AsRef<Path>, S: AsRef<str>>(
     extensions: &[S],
     excludes: &FilePatternSet,
     exclude_mode: ExcludeMode,
+    gitignore_mode: GitignoreMode,
 ) -> Vec<PathBuf> {
     paths
         .iter()
         .flat_map(|path| {
             if matches!(exclude_mode, ExcludeMode::Force) && excludes.matches(path) {
                 vec![]
             } else if path.as_ref().is_dir() {
-                WalkDir::new(path)
-                    .min_depth(1)
-                    .into_iter()
-                    .filter_entry(|e| !excludes.matches(e.path()))
+                let excludes = excludes.clone(); // Needed to get around lifetime issues
+                let mut builder = WalkBuilder::new(path);
+                builder.standard_filters(gitignore_mode.into());
+                builder.hidden(false);
+                builder.filter_entry(move |e| !excludes.matches(e.path()));
+                builder
+                    .build()
                     .filter_map(|p| p.ok()) // skip dirs if user doesn't have permission
                     .filter(|p| is_valid_extension(p.path(), extensions))
```

Comment:
Could use [`TypeBuilder`](https://docs.rs/ignore/0.4.23/ignore/types/struct.TypesBuilder.html) and [`WalkBuilder.types`](https://docs.rs/ignore/0.4.23/ignore/struct.WalkBuilder.html#method.types) to just walk over the requested extensions?

Reply:
I wasn't aware of that feature, I'll take a look.

Reply:
Added in https://github.com/PlasmaFAIR/fortitude/pull/285/commits/5362fa00e9a1aac348af1b4dbf1c2fe329c903db